xonsh/proc.py

```diff
@@ -552,8 +550,6 @@ def __bool__(self):
     def __eq__(self, other):
         return self.returncode == other
 
-    def __hash__(self):
-        return hash(self.returncode)
```

Comment:
I think we want to keep this around.

For me, the ability to do `!(echo "hi") in {1,2,3}` is worth the possibility of hash collisions if someone wants to use these things as keys in a dictionary or store them in a set.  But I'm open to discussion.